packages/uhk-web/src/app/store/reducers/user-configuration.ts

```diff
@@ -28,7 +33,19 @@ export function reducer(state = initialState, action: Action & { payload?: any }
             newKeymap.name = generateName(state.keymaps, newKeymap.name);
             newKeymap.isDefault = (state.keymaps.length === 0);
 
-            changedUserConfiguration.keymaps = state.keymaps.concat(newKeymap);
+            const newKeymaps: Keymap[] = [];
```

Comment:
I think you could create a same function as `addMacro` to create the new keymaps.
But the better solution if you create a generic function instead of `addMacro` and use it everywhere.
```typescript
function createNewElements<T extends Macro | Keymap>(array:T[], newElement:T):T[] {
}
```

it is type safe and 1 code